### PR TITLE
Silence error output in test_concurrent_futures (bpo-21423)

### PR DESCRIPTION
**Pull request by pitrou**
https://bugs.python.org/issue21423

### CODE REVIEW

File: Lib/test/test_concurrent_futures.py [hunk 1]
```diff
@@ -234,14 +253,20 @@ def _prime_executor(self):
...
     def _assert_logged(self, msg):
-        if self.executor_type is futures.ProcessPoolExecutor:
-            # No easy way to catch the child processes' stderr
+        if self.log_queue is not None:
             yield
+            output = []
+            while True:
+                try:
+                    output.append(self.log_queue.get_nowait().getMessage())
+                except queue.Empty:
```
Comment (serhiy-storchaka) [inline Lib/test/test_concurrent_futures.py]: You could catch an exception outside of the loop.

**Review by serhiy-storchaka — DISMISSED** [review]
Excellent! I'm not well known with logging, but it seems to me that this fixes the issue in right way.

File: Lib/test/test_concurrent_futures.py (hunk 1 above)
Comment (pitrou) [reply Lib/test/test_concurrent_futures.py]: Oh, you're right.

**Comment by pitrou** [comment]
I've fixed the tests on Windows now.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**